### PR TITLE
go link: use external linker when in race mode

### PR DESCRIPTION
**Pull request by motiejus**
As of clang 15.0.3 (via zig v0.10), when building with `race = "on"` on x86_64 Linux, we observe the following:

    runtime/cgo(.text): relocation target memset not defined

Using an external linker when in race mode fixes that. Previously `linkmode=external` was set for windows-only race builds, now lets just apply it indiscriminately.

From my past experience, reporting a Go linker error when the external linker works has a high chance to get the ticket closed as unactionable; so it makes sense to just use an external linker, when it works.

> Bug fix

### CODE REVIEW

File: go/private/actions/link.bzl [hunk 1]
```diff
@@ -96,16 +96,18 @@ def emit_link(
         tool_args.add("-msan")
     if ((go.mode.static and not go.mode.pure) or
         go.mode.link != LINKMODE_NORMAL or
-        go.mode.goos == "windows" and (go.mode.race or go.mode.msan)):
+        go.mode.race or
+        go.mode.goos == "windows" and go.mode.msan):
```
Comment (fmeum) [inline go/private/actions/link.bzl:100]:
```suggestion
        (go.mode.goos == "windows" and go.mode.msan)):
```
Makes precedence more obvious

File: go/private/actions/link.bzl [hunk 2]
```diff
@@ -96,16 +96,18 @@ def emit_link(
...
         # * Non-normal build mode: may not be strictly necessary, especially
         #   for modes like "pie".
-        # * Race or msan build for Windows: Go linker has pairwise
-        #   incompatibilities with mingw, and we get link errors in race mode.
-        #   Using the C linker avoids that. Race and msan always require a
-        #   a C toolchain. See #2614.
+        # * Msan build for Windows: Go linker has pairwise incompatibilities
+        #   with mingw. Using the C linker avoids that. Race and msan always
+        #   require a C toolchain. See #2614.
+        # * Race builds (previously Windows, later +linux): we get linker
```
Comment (fmeum) [inline go/private/actions/link.bzl:109]: Let's document the current state only, not the history of this part of the code. We can always rely on git history for that. 

Please also add an example of the error you get with the internal linker.

Comment (motiejus) [reply go/private/actions/link.bzl]: Done, both suggestions.

**Review by fmeum — APPROVED** [review]

**Comment by fmeum** [comment]
@linzhp Looks good to me, I will leave merging to you

**Comment by motiejus** [comment]
Addressed @linzhp 's concerns raised offline.

**Review by fmeum — APPROVED** [review]

**Pull request merged**